tests/virt_autotest/esxi_open_vm_tools.pm

```diff
@@ -81,6 +81,124 @@ sub do_sanity_checks {
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 0, $VM_POWER_OFF);
+
+    record_info('Guest Power On');
+    $powerops = 'power.on';
+    $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_OFF);
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 1, $VM_POWER_ON);
+
+    record_info('Guest Power Reboot');
+    $powerops = 'power.reboot';
+    check_vmtools_service() if (is_svirt);
+    $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_ON);
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 1, $VM_POWER_ON);
+
+    record_info('Guest Power Reset');
+    $powerops = 'power.reset';
+    check_vmtools_service() if (is_svirt);
+    $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_ON);
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 1, $VM_POWER_ON);
+
+    record_info('Guest Power Off');
+    $powerops = 'power.off';
+    check_vmtools_service() if (is_svirt);
+    $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_ON);
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 0, $VM_POWER_OFF);
+
+    # Boot up the VM if it's powered off
+    $powerops = 'power.on';
+    $powerops_ret = take_vm_power_ops($vm_id, $powerops, $VM_POWER_OFF);
+    check_vm_power_state($vm_id, $vm_ip, $powerops, $powerops_ret, 1, $VM_POWER_ON);
+}
+
+sub do_networking_tests {
+    my ($vm_id, $vm_ip) = @_;
+
+    record_info('Networking Tests');
+
+    my $vswitch_name = esxi_vm_network_binding($vm_id);
+    record_info('VSwitch Name', $vswitch_name);
+
+    # Test network via assigned IPv4 address
+    if (is_sle('15+')) {
+        assert_script_run($ssh_vm . "ping -I $vm_ip -4 -c3 " . get_var('OPENQA_URL'));
+        assert_script_run($ssh_vm . "ping -I $vm_ip -4 -c3 www.suse.com");
+    }
+    else {
+        assert_script_run($ssh_vm . "ping -I $vm_ip -c3 " . get_var('OPENQA_URL'));
+        assert_script_run($ssh_vm . "ping -I $vm_ip -c3 www.suse.com");
+    }
+}
+
+sub do_clock_sync_tests {
+    my ($vm_name, $vm_id, $vm_ip) = @_;
+    my ($powerops, $powerops_ret, $g_init_time, $diff_secs);
+
+    record_info('Clock Sync Tests');
+
+    select_console('sut') if (is_svirt && current_console() ne 'sut');
+
+    # Set guest time with the last day of host time before enabling timesync service
+    $g_init_time = init_guest_time();
+    record_info('Guest current time before time sync enabled', $g_init_time);
+
+    record_info('Enable clock sync tests');
+    # Enable timesync service in guest VM
+    if (script_output($ssh_vm . 'vmware-toolbox-cmd timesync status', proceed_on_failure => 1, timeout => 90) eq 'Disabled') {
+        assert_script_run($ssh_vm . 'vmware-toolbox-cmd timesync enable', timeout => 90);
+    }
+
+    # Check the time difference between host and guest after timesync service enabled
+    $diff_secs = get_diff_seconds();
+
+    if ($diff_secs < 10) {
```

Comment:
Why 10 is a reasonable value ?

Reply:
Pls refer to the PR#13036, I didn't make any changes in test subroutines, just adjusting the sequence of subroutines.